pandas/core/indexes/base.py

```diff
@@ -53,7 +53,7 @@
     ABCDataFrame,
     ABCDatetimeArray,
     ABCDatetimeIndex,
-    ABCIndexClass,
```

Comment:
Why was this removed?

Reply:
Since `Index` implements `ABCIndexClass`, the check on line 516 will be faster if we just use `Index` itself. Since thats the only instance, we can then remove this. It's a small tradeoff I can put back if you would prefer to keep using ABCIndexClass here.

Basically my understanding is if something actually requires Index in the file we should check instances with Index directly, otherwise we should import ABCIndexClass to save on importing a larger file.

Reply:
Hmm it's a good thought but I'd maybe leave that for a separate discussion to address more comprehensively

Reply:
Alright, posed an update.